### PR TITLE
filterable power node connections

### PR DESCRIPTION
**Pull request by KayyAyeAre**
If your pull request is **not** translation or serverlist-related, read the list of requirements below and check each box:

- [x] I have read the [contribution guidelines](https://github.com/Anuken/Mindustry/blob/master/CONTRIBUTING.md).
- [x] I have ensured that my code compiles, if applicable.
- [x] I have ensured that any new features in this PR function correctly in-game, if applicable.

### CODE REVIEW

**Comment by KayyAyeAre** [comment]
tested this by overriding `shouldConnect()` in the mend projector
image

**Comment by GlennFolker** [comment]
I feel like, for the sake of the usefulness, other parameters should be passed as well. `boolean shouldConnect(Tile connector, Block connectorType, Building build)` would probably be sufficient.

> other.block.shouldConnect(other)

Turns to `other.block.shouldConnect(tile, this, other)`, and you should probably add to the document that `tile.build` might be `null` there.

**Comment by MEEPofFaith** [comment]
(`@Nullable`)

**Comment by bendnuts** [comment]
all this seems to do is change 2 lines of code... is it supposed to make it so mods can make content not automatically connect to power nodes?

File: core/src/mindustry/world/blocks/power/PowerNode.java [hunk 1]
```diff
@@ -205,6 +205,7 @@ protected void getPotentialLinks(Tile tile, Team team, Cons<Building> others){
             overlaps(tile.x * tilesize + offset, tile.y * tilesize + offset, other.tile(), laserRange * tilesize) && other.team == team &&
             !graphs.contains(other.power.graph) &&
             !PowerNode.insulated(tile, other.tile) &&
+            other.block.shouldConnect(other) &&
```
Comment (Mnemotechnician) [inline core/src/mindustry/world/blocks/power/PowerNode.java:208]: I may not understand something , but why are you checking whether a block should connect to itself?

**Comment by Anuken** [comment]
I agree with Glenn here; there aren't enough parameters. It's also in a strange file - if something requires its own building, it should be in BuildingComp, not Block. In case you still need this, re-open with the changes.

**Pull request closed without merging**